### ISSUE TITLE
networkd drops static IP when renewing DHCP lease

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#16157]
**systemd version the issue has been seen with**

v241

**Used distribution**

Raspbian

I have the following networkd configuration on one of my RPIs, which is supposed to set a static IP address (192.168.178.11) and get a second one via DHCP (192.168.178.21).
```
# cat /etc/systemd/network/25-wireless.network 
[Match]
Name=wlan0

[Network]
DHCP=ipv4
Address=192.168.178.11/24
```

This works fine when networkd is started :
```
3: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether XX:XX:XX:XX:XX:XX brd ff:ff:ff:ff:ff:ff
    inet 192.168.178.11/24 brd 192.168.178.255 scope global wlan0
       valid_lft forever preferred_lft forever
    inet 192.168.178.21/24 brd 192.168.178.255 scope global secondary dynamic wlan0
       valid_lft 863387sec preferred_lft 863387sec
```
But whenever a DHCP lease is renewed, networkd removes the static `192.168.178.11` and does not reapply it.

A relevant excerpt from the debug log:

```
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Flags change: -RUNNING
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=18 reply_cooki
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Lost carrier
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCP CLIENT (0x63f3dac4): STOPPED
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: DHCP lease lost
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address 192.168.178.21
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: DHCPv6 lease lost
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: NDISC: Stopping IPv6 Router Solicitation client
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address 192.168.178.11
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address XXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:XXXX
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: State is configured, dropping config
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address XXXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:YYYY
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address 192.168.178.21
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=19 reply_cookie=0 signa
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Flags change: +RUNNING
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=20 reply_cooki
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Gained carrier
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Acquiring DHCPv4 lease
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCP CLIENT (0x480b4c36): STARTED on ifindex 3
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Discovering IPv6 routers
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: NDISC: Started IPv6 Router Solicitation client
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=21 reply_cooki
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Setting addresses
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=22 reply_cookie=0 signa
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: NDISC: Sent Router Solicitation, next solicitation in 3s
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCP CLIENT (0x480b4c36): REQUEST (init-reboot)
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCP CLIENT (0x480b4c36): ACK
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCP CLIENT (0x480b4c36): lease expires in 1w 2d 23h 59min 59s
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCP CLIENT (0x480b4c36): T2 expires in 1w 1d 17h 59min 57s
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCP CLIENT (0x480b4c36): T1 expires in 4d 23h 59min 58s
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: DHCPv4 address 192.168.178.21/24 via 192.168.178.1
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: NDISC: Received Router Advertisement: flags OTHER preference medium lifetime 1800 sec
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: NDISC: Invoking callback for 'router' event.
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Started in Information request mode
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Acquiring DHCPv6 lease on NDisc request
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Sent INFORMATION-REQUEST
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Next retransmission in 998ms
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address: 192.168.178.21/24 (valid for 1w 2d 19h 4min)
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address: 192.168.178.11/24 (valid forever)
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address: XXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:XXXX/64 (valid for 1h 52min 52s)
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Recv REPLY
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: wlan0: Removing address: XXXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:YYYY/64 (valid for 1h 52min 52s)
Jun 12 09:52:14 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=23 reply_cooki
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=24 reply_cookie=0 signa
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Adding address: 192.168.178.11/24 (valid forever)
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=25 reply_cooki
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=26 reply_cookie=0 signa
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Addresses set
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Adding address: 192.168.178.21/24 (valid for 1w 3d)
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: DHCP: No routes received from DHCP server: No data available
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Adding address: XXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:XXXX/64 (valid for 2h)
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Configured
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=27 reply_cooki
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Removing address: 192.168.178.11/24 (valid forever)
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Updating address: 192.168.178.21/24 (valid for 1w 3d)
Jun 12 09:52:15 raspberrypi systemd-networkd[14790]: wlan0: Adding address: XXXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:YYYY/64 (valid for 2h)
Jun 12 09:52:16 raspberrypi systemd-networkd[14790]: wlan0: Updating address: XXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:XXXX/64 (valid for 1h 59min 59s)
Jun 12 09:52:17 raspberrypi systemd-networkd[14790]: wlan0: Updating address: XXXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:YYYY/64 (valid for 1h 59min 58s)
Jun 12 09:52:26 raspberrypi systemd-networkd[14790]: NDISC: No RA received before link confirmation timeout
Jun 12 09:52:26 raspberrypi systemd-networkd[14790]: NDISC: Invoking callback for 'timeout' event.
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: NDISC: Received Router Advertisement: flags OTHER preference medium lifetime 1800 sec
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: NDISC: Invoking callback for 'router' event.
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Started in Information request mode
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: wlan0: Acquiring DHCPv6 lease on NDisc request
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: wlan0: Updating address: XXXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:YYYY/64 (valid for 2h)
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Sent INFORMATION-REQUEST
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Next retransmission in 919ms
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: wlan0: Updating address: XXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:XXXX/64 (valid for 2h)
Jun 12 09:53:06 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Recv REPLY
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: NDISC: Received Router Advertisement: flags OTHER preference medium lifetime 1800 sec
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: NDISC: Invoking callback for 'router' event.
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Started in Information request mode
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: wlan0: Acquiring DHCPv6 lease on NDisc request
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: wlan0: Updating address: XXXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:YYYY/64 (valid for 2h)
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Sent INFORMATION-REQUEST
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Next retransmission in 1s
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: DHCPv6 CLIENT: Recv REPLY
Jun 12 09:53:11 raspberrypi systemd-networkd[14790]: wlan0: Updating address: XXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:XXXX/64 (valid for 2h)
```

**Comment by ssahani** [comment]
I cant reproduce this with current git.
```
 cat veth99.network
[Match]
Name=veth99

[Network]
IPv6AcceptRA=no
DHCP=ipv4
Address=192.168.4.5

```
```
eth99: Remembering updated address: 192.168.5.35/24 (valid for 20s)
veth99: DHCP: No routes received from DHCP server: No data available
veth99: Configuring route: dst: 192.168.5.1/32, src: n/a, gw: n/a, prefsrc: 192.168.5.35, scope: link, table: main, proto: dhcp, type: unicast
veth99: Configuring route: dst: n/a, src: n/a, gw: 192.168.5.1, prefsrc: 192.168.5.35, scope: global, table: main, proto: dhcp, type: unicast
rtnl: received non-static neighbor, ignoring.
NDISC: Sent Router Solicitation, next solicitation in 2min 19s
rtnl: received non-static neighbor, ignoring.
rtnl: received non-static neighbor, ignoring.
rtnl: received non-static neighbor, ignoring.
rtnl: received non-static neighbor, ignoring.
DHCP CLIENT (0x2d3dc317): REQUEST (renewing)
rtnl: received non-static neighbor, ignoring.
rtnl: received non-static neighbor, ignoring.
DHCP CLIENT (0x2d3dc317): REQUEST (rebinding)
DHCP SERVER: REQUEST (rebinding/renewing) (0x2d3dc317)
DHCP SERVER: ACK (0x2d3dc317)
DHCP CLIENT (0x2d3dc317): ACK
DHCP CLIENT (0x2d3dc317): lease expires in 18s
DHCP CLIENT (0x2d3dc317): T2 expires in 15s
DHCP CLIENT (0x2d3dc317): T1 expires in 9s
veth99: Remembering updated address: 192.168.5.35/24 (valid for 20s)
veth99: DHCP: No routes received from DHCP server: No data available
veth99: Configuring route: dst: 192.168.5.1/32, src: n/a, gw: n/a, prefsrc: 192.168.5.35, scope: link, table: main, proto: dhcp, type: unicast
veth99: Configuring route: dst: n/a, src: n/a, gw: 192.168.5.1, prefsrc: 192.168.5.35, scope: global, table: main, proto: dhcp, type: unicast
rtnl: received non-static neighbor, ignoring.



```

**Comment by yuwata** [comment]
Yeah, I neither reproduce this with current git master. Let's close this.

**Comment by mbiebl** [comment]
I can reproduce this with v245.6, so I guess this should be reopened.

How can I provide you with more information?

Am Mi., 17. Juni 2020 um 07:04 Uhr schrieb Yu Watanabe <
notifications@github.com>:

**Comment by mbiebl** [comment]
I switched the underlying hardware (from a RPI 1 to RPI 3) and I no longer encounter the issue (I've been running the new system for a couple of days now).
I just switched the SD card, so the software configuration is identical.
Not sure what to do with this issue now. Maybe it's just a race condition which I no longer hit since the RPI 3 is much faster then the older model.

Let's close this bug report at this point.
